### DISCUSSION TITLE
verification of .cab file during offline udpate.

### DISCUSSION THREAD

**Discussion by Aravindh09** [fwupd/fwupd#3487] [Q&A] [answered]
Hi,

is there a way to verify the .cab file downloaded from LVFS site during an offline update?
my doubt is do we have any command line parameter that we can use with fwupdmgr verify-signature <filename.cab> which will say signature verification successful or signature verification failed for the given <filename.cab> file.

**Comment by hughsie** [comment, marked as answer]
Can you do something like:

	[hughsie@sr630v2 fwupd (master *%)]$ fwupdmgr get-details firmware.cab 
	Decompressing…           [***************************************]
	20QD0019JP
	│
	└─Unknown Device:
	  │   Description:        
	  │   Lenovo ThinkPad P50 System Firmware N1EET91W, version 1.64
	  │   GUID:               ddc0ee61-e7f0-4e7d-acc5-c070a398838e
	  │ 
	  └─ThinkPad P50:
		New version:      0.1.64
		Summary:          Lenovo ThinkPad P50 System Firmware
		Licence:          Proprietary
		Urgency:          High
		Vendor:           Lenovo Ltd.
		Flags:            trusted-payload|trusted-metadata
		Description:      
		Lenovo ThinkPad P50 System Firmware Version 1.64
		
		Important updates:
		
		• Addresses CVE issues.
		
		New functions or enhancements:
		
		• Updated the CPU microcode.
		• Updated the Diagnostics module to version 04.17.000.

The `trusted-payload|trusted-metadata` tells you it is trusted by one of the keys you have locally installed.

**Reply by Aravindh09** [reply]
Hi @hughsie Thank you so much for your response. 
One more doubt that i have is are all .cab files uploaded to the LVFS by OEM signed before being hosted and made available for download by the public ? Basically can i end up downloading an unsigned .cab file from LVFS? if all files hosted on LVFS is signed are they signed using LVFS GPG/PKCS#7 private key ?

Hoping for your response. 
Thank you.

**Reply by hughsie** [reply]
The .cab files are signed by the LVFS (using gpg2/pkcs#7) so you can know that they actually originated from the LVFS. The OEM may sign the payload, but we don't have any visibility on that.

**Reply by Aravindh09** [reply]
Thank you @hughsie. 
Information you provided helps a lot. 
:)
